src/network_event/client_event.rs

```diff
@@ -119,155 +119,219 @@ impl ClientEventAppExt for App {
         &mut self,
         channel: impl Into<RepliconChannel>,
     ) -> &mut Self {
-        self.add_client_event_with::<T, _, _>(channel, send::<T>, receive::<T>)
+        self.add_client_event_with::<T>(channel, send::<T>, receive::<T>)
     }
 
     fn add_mapped_client_event<T: Event + Serialize + DeserializeOwned + MapEntities + Clone>(
         &mut self,
         channel: impl Into<RepliconChannel>,
     ) -> &mut Self {
-        self.add_client_event_with::<T, _, _>(channel, map_and_send::<T>, receive::<T>)
+        self.add_client_event_with::<T>(channel, map_and_send::<T>, receive::<T>)
     }
 
-    fn add_client_event_with<T: Event, Marker1, Marker2>(
+    fn add_client_event_with<T: Event + Serialize + DeserializeOwned>(
         &mut self,
         channel: impl Into<RepliconChannel>,
-        send_system: impl IntoSystemConfigs<Marker1>,
-        receive_system: impl IntoSystemConfigs<Marker2>,
+        send_fn: SendFn,
+        receive_fn: ReceiveFn,
     ) -> &mut Self {
         let channel_id = self
             .world
             .resource_mut::<RepliconChannels>()
             .create_client_channel(channel.into());
 
         self.add_event::<T>()
-            .init_resource::<Events<FromClient<T>>>()
-            .insert_resource(ClientEventChannel::<T>::new(channel_id))
-            .add_systems(
-                PreUpdate,
-                (
-                    reset::<T>.in_set(ClientSet::ResetEvents),
-                    receive_system
-                        .in_set(ServerSet::Receive)
-                        .run_if(server_running),
-                ),
-            )
-            .add_systems(
-                PostUpdate,
-                (
-                    send_system.run_if(client_connected),
-                    resend_locally::<T>.run_if(has_authority),
-                )
-                    .chain()
-                    .in_set(ClientSet::Send),
-            );
+            .init_resource::<Events<FromClient<T>>>();
+
+        self.world
+            .get_resource_mut::<ClientEventRegistry>()
+            .unwrap()
+            .register_event::<T>(channel_id, send_fn, receive_fn);
 
         self
     }
 }
 
-fn receive<T: Event + DeserializeOwned>(
-    mut client_events: EventWriter<FromClient<T>>,
-    mut server: ResMut<RepliconServer>,
-    channel: Res<ClientEventChannel<T>>,
-) {
-    for (client_id, message) in server.receive(*channel) {
-        match DefaultOptions::new().deserialize(&message) {
-            Ok(event) => {
-                trace!(
-                    "applying event `{}` from `{client_id:?}`",
-                    any::type_name::<T>()
-                );
-                client_events.send(FromClient { client_id, event });
-            }
-            Err(e) => debug!("unable to deserialize event from {client_id:?}: {e}"),
+fn map_and_send<T: Event + MapEntities + Serialize + Clone>(world: &mut World, channel: u8) {
```

Comment:
Could you put `send` first? Just a small personal preference :)

Reply:
I moved it below the `send` function